File: BE/SSAIREN/src/main/java/com/ssairen/domain/file/dto/FileUploadResponse.java
package com.ssairen.domain.file.dto;

import lombok.Builder;
import lombok.Getter;

/**
 * 파일 업로드 응답 DTO
 */
@Getter
@Builder
public class FileUploadResponse {

    /**
     * 업로드된 파일의 고유 이름 (UUID 기반)
     */
    private String fileName;

    /**
     * 원본 파일명
     */
    private String originalFileName;

    /**
     * 파일 크기 (바이트)
     */
    private Long fileSize;

    /**
     * 파일 콘텐츠 타입 (MIME type)
     */
    private String contentType;

    /**
     * 파일이 저장된 버킷 이름
     */
    private String bucketName;

    /**
     * 파일 URL (MinIO에서 직접 접근 가능한 URL)
     */
    private String fileUrl;
}

File: FE/ssairen_app/app/src/main/java/com/example/ssairen_app/data/dto/FileUploadResponse.kt
package com.example.ssairen_app.data.dto

import com.google.gson.annotations.SerializedName

/**
 * 파일 업로드 응답 DTO
 * 백엔드 FileUploadResponse와 동일한 구조
 */
data class FileUploadResponse(
    @SerializedName("fileName")
    val fileName: String,

    @SerializedName("originalFileName")
    val originalFileName: String,

    @SerializedName("fileSize")
    val fileSize: Long,

    @SerializedName("contentType")
    val contentType: String,

    @SerializedName("bucketName")
    val bucketName: String,

    @SerializedName("fileUrl")
    val fileUrl: String
)
